src/UiPath.CoreIpc.Tests/NamedPipeTests.cs

```diff
@@ -43,6 +43,29 @@ public async Task PipeSecurityForWindows()
             _ = protectedService.RunAsync();
             await CreateSystemService().DoNothing().ShouldThrowAsync<UnauthorizedAccessException>();
         }
+
+        [Fact]
+        public async Task PipeCancelIoOnServer()
+        {
+            (await _systemClient.CancelIoPipe(new())).ShouldBeTrue();
+
+            (await _systemClient.Delay()).ShouldBeTrue();
+        }
+
+        [Fact]
+        public async Task PipeCancelIoOnClient()
+        {
+            (await _systemClient.Delay()).ShouldBeTrue();
+
+            var delayTask = _systemClient.Delay(500);
+            await Task.Delay(100);
+            var pipeStream = ((IpcProxy)_systemClient).Connection.Network as PipeStream;
+            SystemService.CancelIoEx(pipeStream.SafePipeHandle.DangerousGetHandle(), IntPtr.Zero).ShouldBeTrue();
+
+            (await delayTask).ShouldBeTrue();
+
+            (await _systemClient.Delay()).ShouldBeTrue();
+        }
```

Comment:
what are the Delay(0) calls doing in the tests? is it like DoNothing to assert that Ipc still works?

Reply:
Yes, the assert is the Ipc still works. I prefer to use Delay instead of DoNothing (that is fire and forget)